### PR TITLE
Docker3 ja

### PR DESCRIPTION
**Pull request by geekscruff**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,10 +6,13 @@ ARG RAILS_ENV
 ARG DERIVATIVES_PATH
 ARG UPLOADS_PATH
 ARG CACHE_PATH
+ARG WORKING_PATH
```
Comment (anusharanganathan) [inline Dockerfile]: This doesn't seem to be used

File: Dockerfile [hunk 2]
```diff
@@ -30,34 +33,37 @@ RUN echo 'deb http://deb.debian.org/debian jessie-backports main' > /etc/apt/sou
...
-RUN /bin/repo_builder.sh
+RUN /bin/repo_builder.sh -gemkey $GEM_KEY -gemsrc $GEM_SOURCE
 
-COPY setup.sh /bin/
-RUN chmod +x /bin/setup.sh
-RUN /bin/setup.sh
-
-RUN mkdir -p /app/hull_culture/shared/pids
-RUN mkdir -p /app/hull_culture/shared/log/hull_culture
+RUN bundle install --without development test
```
Comment (anusharanganathan) [inline Dockerfile]: Should we check for RAILS_ENV here?

File: Dockerfile [hunk 3]
```diff
@@ -30,34 +33,37 @@ RUN echo 'deb http://deb.debian.org/debian jessie-backports main' > /etc/apt/sou
...
+RUN bundle clean --force
```
Comment (anusharanganathan) [inline Dockerfile]: and here

File: repo_builder.sh [hunk 4]
```diff
@@ -1,27 +1,22 @@
...
-export GEM_SOURCE=https://github.com/research-technologies
-export APPLICATION_KEY=hull_culture
-export APP_URL='https://github.com/research-technologies/hyrax_leaf'
-export BRANCH='master'
+while getopts gemkey:gemsrc: option; do
+  case ${option} in
+  gemkey) GEM_KEY=${OPTARG};;
+  gemsrc) GEM_SOURCE=${OPTARG};;
+  esac
+done
```
Comment (anusharanganathan) [inline repo_builder.sh]: If you set these env variables in .env, wouldn't they just be available to this file without having to pass the args and read them?

File: Dockerfile (hunk 1 above)
Comment (geekscruff) [reply Dockerfile]: It is used by Hyrax I think, see:

https://github.com/research-technologies/hyrax_leaf/pull/5/files#diff-39ac5326a57fe21da35b98b056174a3dR14
and
https://github.com/research-technologies/hyrax_leaf/pull/5/files#diff-39ac5326a57fe21da35b98b056174a3dR14

It's seems like there's a lot of these paths, but I thought it useful to separate them out.

File: Dockerfile (hunk 3 above)
Comment (geekscruff) [reply Dockerfile]: done :)

File: repo_builder.sh (hunk 4 above)
Comment (geekscruff) [reply repo_builder.sh]: oh, yes ... and I thought I was being so clever with my params!

File: Dockerfile (hunk 1 above)
Comment (geekscruff) [reply Dockerfile]: I create one volume (uploads) and use it for uploads/cache/working_dir

Comment (anusharanganathan) [reply Dockerfile]: Thanks for telling about working_dir. I didn't know that one

**Pull request merged**